CMakeLists.txt

```diff
@@ -312,6 +315,17 @@ if(BUILD_TESTING)
     target_link_libraries(test_repl_str ${PROJECT_NAME})
   endif()
 
+  set(append_library_dirs "$<TARGET_FILE_DIR:${PROJECT_NAME}>")
+
+  ament_add_gtest(test_shared_library test/test_shared_library.cpp
+    APPEND_LIBRARY_DIRS "${append_library_dirs}"
+  )
+
+  if(TARGET test_shared_library)
+    add_library(dummy_shared_library test/dummy_shared_library/dummy_shared_library.c)
```

Comment:
Here is where you'd need to define `DUMMY_SHARED_LIBRARY_BUILDING_DLL`..

Reply:
thank you @wjwwood https://github.com/ros2/rcutils/pull/215/commits/66187a37d0db7787aa20d48104b451dc8a10f6cb